dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/secrets/DotVelocitySecretAppConfig.java

```diff
@@ -0,0 +1,158 @@
+    }
+
+    public final String[] getArrayValue(final String key) {
+
+        return UtilMethods.isSet(extraParameters.get(key))
+                        ? Arrays.stream(extraParameters.get(key).split(",")).map(String::trim).toArray(String[]::new)
+                        : new String[0];
+    }
+
+    private DotVelocitySecretAppConfig(Builder builder) {
+        this.title = builder.title;
+        this.extraParameters = builder.extraParameters;
+    }
+
+    public static Optional<DotVelocitySecretAppConfig> config() {
+
+        return config(HttpServletRequestThreadLocal.INSTANCE.getRequest());
+    }
+
+    /**
+     * Gets the secrets from the App - this will check the current host then the SYSTEM_HOST for a valid
+     * configuration. This lookup is low overhead and cached by dotCMS.
+     * 
+     * @param request {@link HttpServletRequest}
+     * @return
+     */
+    public static Optional<DotVelocitySecretAppConfig> config(final HttpServletRequest request) {
+
+        return config(null != request? WebAPILocator.getHostWebAPI().getCurrentHostNoThrow(request) : APILocator.systemHost());
+    }
+
+    /**
+     * Gets the secrets from the App - this will check the current host then the SYSTEM_HOST for a valid
+     * configuration. This lookup is low overhead and cached by dotCMS.
+     *
+     * @param host {@link Host}
+     * @return
+     */
+    public static Optional<DotVelocitySecretAppConfig> config(final Host host) {
+
+        if (DotVelocitySecretAppConfigThreadLocal.INSTANCE.getConfig(host.getIdentifier()).isPresent()) {
+
+            return DotVelocitySecretAppConfigThreadLocal.INSTANCE.getConfig(host.getIdentifier());
+        }
+
+        final Optional<AppSecrets> appSecrets = Try.of(
+                        () -> APILocator.getAppsAPI().getSecrets(DotVelocitySecretAppKeys.APP_KEY, true, host, APILocator.systemUser()))
+                .getOrElse(Optional.empty());
+
+        if (!appSecrets.isPresent()) {
+
+            return Optional.empty();
+        }
+
+        final Map<String, Secret> secrets = new HashMap<>(appSecrets.get().getSecrets());
+        final Map<String, String> extraParameters = new HashMap<>();
+        final String title =
+                Try.of(() -> secrets.get(DotVelocitySecretAppKeys.TITLE.key).getString()).getOrNull();
+
+        if (!UtilMethods.isSet(title)) {
+
+            return Optional.empty();
+        }
+
+        for (String key : secrets.keySet()) {
+            extraParameters.put(key, secrets.get(key).getString());
+        }
+
+        final DotVelocitySecretAppConfig config = DotVelocitySecretAppConfig.builder()
+                .withTitle(title)
+                .withExtraParameters(extraParameters).build();
+
+        DotVelocitySecretAppConfigThreadLocal.INSTANCE.setConfig(host.getIdentifier(), Optional.ofNullable(config));
+
+        return DotVelocitySecretAppConfigThreadLocal.INSTANCE.getConfig(host.getIdentifier());
+    }
+
```

Comment:
Shouldn't we always use `AppSecrets` as the carrier of secrets and the call destroy once we're done? Here you're copying the secrets for convenience in strings and as they are immutable the reference sticks around in the VM.
I would try to stick to the  AppSecrets and always call destroy once it has been used.